lib/ocpp/v201/smart_charging.cpp

```diff
@@ -21,6 +21,19 @@ ProfileValidationResultEnum SmartChargingHandler::validate_evse_exists(int32_t e
                                               : ProfileValidationResultEnum::Valid;
 }
 
+ProfileValidationResultEnum
+SmartChargingHandler::validate_charge_point_max_profile(const ChargingProfile& profile) const {
+    if (profile.chargingProfilePurpose != ChargingProfilePurposeEnum::ChargingStationMaxProfile) {
+        return ProfileValidationResultEnum::InvalidProfileType;
+    }
```

Comment:
Per the spec, `ChargingStationMaxProfile` can only be set on EVSE ID 0 (section 3.2, page 238). Why are we not checking for that in the validation? And why do we assume that the only purpose can be `MaxProfile`?

Reply:
This is a really good callout. I was only looking at the specific reqs from K01. Updated the method signature to match what's being done for `validate_tx_profile()` and added the test `K04FR04_ValidateChargingStationMaxProfile_EvseIDgt0_Invalid`.